src/libsyntax/ast.rs

```diff
@@ -924,6 +925,13 @@ pub struct Expr {
     pub attrs: ThinVec<Attribute>,
 }
 
+// This type is used a lot. Make sure it doesn't get bigger. (And if it gets
+// smaller, this assertion should be updated so it stays at that smaller size.)
+// Note that the assertion is x86_64-only because the size is larger on
+// AArch64.
+#[cfg(target_arch = "x86_64")]
+static_assert!(EXPR_IS_88_BYTES: std::mem::size_of::<Expr>() == 88);
```

Comment:
It seems a bit incongruous that this assert is `==` whereas the Statement assert is `<=` -- I would think the same logic should apply to both.

Reply:
The `Statement` assert is used on all platforms, which is why it's looser -- the size on 32-bit is different to the size on 64-bit, etc.

The `Expr` assert is x86_64-only, because the size on AArch64 is apparently greater than 88, and I didn't want to give unnecessary leeway on x86_64, which is probably the most important platform. So I took the opportunity to make it stronger, using equality. The advantage of this is that if somebody does shrink `Expr`, say to 80 bytes, they must update the assertion accordingly, and then we can't accidentally backslide back to 88 bytes. That's what the "And if it gets smaller, this assertion should be updated so it stays at that smaller size" part of the comment is trying to communicate.

So, options:
- Accept the code as is.
- Expand comments on one or both assertions.
- Make the assertions more consistent, possibly by restricting the `Statement` assertion to x86_64-only.

@Mark-Simulacrum: which would you prefer?

Reply:
I think it probably makes sense to gate the Statement assert to x86_64 as well and make it ==.